core/src/main/java/org/apache/calcite/rel/core/JoinRelType.java

```diff
@@ -94,6 +150,37 @@ public JoinRelType cancelNullsOnRight() {
       return this;
     }
   }
+
+  /** Transform this JoinRelType to CorrelateJoinType. **/
+  public CorrelateJoinType toLinq4j() {
+    switch (this) {
+    case INNER:
+      return CorrelateJoinType.INNER;
+    case LEFT:
+      return CorrelateJoinType.LEFT;
+    case SEMI:
+      return CorrelateJoinType.SEMI;
+    case ANTI:
+      return CorrelateJoinType.ANTI;
+    }
+    throw new IllegalStateException(
+        "Unable to convert " + this + " to CorrelateJoinType");
+  }
+
+  public boolean projectsRight() {
```

Comment:
Why not keeping the `returnsJustFirstInput` name from the (now deprecated) SemiJoinType for this new method (the logic is the same in both)?

Reply:
@rubenada Julian suggests this name, you can see the details in the JIRA

Reply:
Ok, sorry I hadn't seen that